lib/bulk-apply-applications.js

```diff
@@ -243,86 +249,143 @@ function bulkApplyApplications (args, callback) {
+            if (applications[0].status === 'approved' || applications[0].status === 'pending' || applications[0].status === 'cancelled' ) {
+              emailParents(cb);
+            } else {
+              return cb();
+            }
+          });
 
-            function emailParents (cb) {
-              if (_.isEmpty(profile.parents)) return cb();
-              var parentsEmailed = [];
-              async.eachSeries(profile.parents, function (parent, cb) {
-                if (!_.isObject(parent)) {
-                  seneca.act({role: 'cd-users', cmd: 'load', id: parent, user: requestingUser}, function (err, parentUser) {
-                    if (err) return cb(err);
-                    payload.to = parentUser.email;
-                    if (!_.contains(parentsEmailed, payload.to) && payload.to !== profile.email) {
-                      parentsEmailed.push(payload.to);
-                      seneca.act({role: 'cd-dojos', cmd: 'send_email', payload: payload}, cb);
-                    } else {
-                      return cb();
-                    }
-                  });
-                } else {
-                  payload.to = parent.email;
+          //email the parent email address
+          function emailParents (cb) {
+            if (_.isEmpty(profile.parents)) return cb();
+            var parentsEmailed = [];
+            async.eachSeries(profile.parents, function (parent, cb) {
+              if (!_.isObject(parent)) {
+                seneca.act({role: 'cd-users', cmd: 'load', id: parent, user: requestingUser}, function (err, parentUser) {
+                  if (err) return cb(err);
+                  payload.to = parentUser.email;
                   if (!_.contains(parentsEmailed, payload.to) && payload.to !== profile.email) {
                     parentsEmailed.push(payload.to);
+                    //send the email
                     seneca.act({role: 'cd-dojos', cmd: 'send_email', payload: payload}, cb);
                   } else {
                     return cb();
                   }
+                });
+              } else {
+                payload.to = parent.email;
+                if (!_.contains(parentsEmailed, payload.to) && payload.to !== profile.email) {
+                  parentsEmailed.push(payload.to);
+                  //send the email
+                  seneca.act({role: 'cd-dojos', cmd: 'send_email', payload: payload}, cb);
+                } else {
+                  return cb();
                 }
-              }, cb);
+              }
+            }, cb);
+          }
+        });
+      });
+      return done(null, profiles, event, session, dojo, tickets, emailCode, dojoEmailCode);
+    }
+
+    function sendDojoEmail (profiles, event, session, dojo, tickets, emailCode, dojoEmailCode, done) {
+      process.nextTick(function () {
+        //if email notifications are enabled for the event, send an email to the dojo
+        if (eventData.emailNotifications) {
```

Comment:
Dojo will receive notification for every type of action? approval required, booked ..?